### PR TITLE
Change create app page path, breadcrumb fix

### PR DESCRIPTION
**Pull request by peterserfozo**

### CODE REVIEW

File: src/Breadcrumb/CreateAppForDeveloperBreadcrumbBuilder.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  public function build(RouteMatchInterface $route_match) {
+    $breadcrumb = new Breadcrumb();
+    $user = $route_match->getParameter('user');
+
+    $breadcrumb->addLink(Link::createFromRoute($this->t('Home'), '<front>'));
+    $breadcrumb->addLink(Link::createFromRoute(User::load($user)->getUsername(), 'entity.user.canonical', [
+      'user' => $user,
+    ]));
+    $breadcrumb->addLink(Link::createFromRoute(
+      t('My @developer_app', [
```
Comment (mxr576) [inline src/Breadcrumb/CreateAppForDeveloperBreadcrumbBuilder.php]: Couldn't you leverage the `_title_callback` of the `entity.developer_app.collection_by_developer` route somehow here? This seems a code duplication to me and if the default page title changes in `\Drupal\apigee_edge\Controller\DeveloperAppListBuilderForDeveloper::getPageTitle` or the `_title_callback` of the route gets altered then the title of the page and what a user can see in the breadcrumb is not going to be aligned.

File: tests/src/Functional/DeveloperAppUITestTrait.php [hunk 2]
```diff
@@ -133,7 +137,14 @@ protected function postCreateAppForm(array $data, ?UserInterface $account = NULL
       $account = $this->account;
     }
 
-    $this->drupalPostForm("/user/{$account->id()}/apps/create", $data, 'Add developer app');
+    $this->drupalPostForm(
```
Comment (mxr576) [inline tests/src/Functional/DeveloperAppUITestTrait.php:142]: Please add a test that ensures the custom breadcrumb is correct on the Add app form, this test should also validate if the custom breadcrumb works properly for different users (breadcrumb cache issue).

File: apigee_edge.services.yml [hunk 3]
```diff
@@ -48,6 +48,11 @@ services:
     tags:
       - { name: event_subscriber }
 
+  apigee_edge.breadcrumb.create_app:
+    class: Drupal\apigee_edge\Breadcrumb\CreateAppForDeveloperBreadcrumbBuilder
+    tags:
+    - { name: breadcrumb_builder, priority: 1001 }
```
Comment (mxr576) [inline apigee_edge.services.yml]: Based on a quick query on the core and open social 101 should be enough as a default priority, we do not need to be this drastic :D

**Review by mxr576 — CHANGES_REQUESTED** [review]

File: apigee_edge.module [hunk 4]
```diff
@@ -1353,3 +1369,23 @@ function _apigee_edge_set_sync_in_progress(?bool $in_progress = NULL) {
...
+ * Gets the title of the my developer apps page.
+ *
+ * @param int $user
+ *   User id.
+ *
+ * @return \Drupal\Core\StringTranslation\TranslatableMarkup
+ *   Title of the my developer apps page.
+ */
+function _apigee_edge_get_my_apps_title($user = NULL) {
+  $args['@developer_app'] = \Drupal::entityTypeManager()->getDefinition('developer_app')->getPluralLabel();
```
Comment (mxr576) [inline apigee_edge.module]: Please introduce a new alter hook that will allow to change the title of the My apps page consistently in all places where this function gets called. `_` wont't be needed anymore.

**Review by mxr576 — CHANGES_REQUESTED** [review]
Please fix the mentioned issue in my new comment.

File: apigee_edge.api.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Alters the title of "My developer apps" page and menu link item.
+ *
+ * @param \Drupal\Core\StringTranslation\TranslatableMarkup $title
+ *   The menu link/page title.
+ * @param \Drupal\user\UserInterface|null $user
+ *   It can be used if an admin visits another user's
+ *   "My developer apps" page.
+ */
+function hook_my_developer_apps_title_alter(TranslatableMarkup &$title, ?UserInterface $user = NULL) {
```
Comment (mxr576) [inline apigee_edge.api.php]: hook_apigee_edge_my....

File: apigee_edge.api.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+/**
+ * @file
+ * Hooks for apigee_edge module.
+ */
+
+use Drupal\Core\StringTranslation\TranslatableMarkup;
```
Comment (mxr576) [inline apigee_edge.api.php]: Always use the fully qualified class names in parameters, do not import them. See node.api.php (Let me know if it is not clear why).

File: tests/modules/apigee_edge_test/apigee_edge_test.module [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_my_developer_apps_title_alter().
+ *
+ * Tests "My developer apps" title alter hook.
+ *
+ * @see \Drupal\Tests\apigee_edge\Functional\DeveloperAppUITest
+ */
+function apigee_edge_test_my_developer_apps_title_alter(TranslatableMarkup &$title, ?UserInterface $user = NULL) {
```
Comment (mxr576) [inline tests/modules/apigee_edge_test/apigee_edge_test.module]: Do not forget to fix the name of the hook implementation here either :)

File: apigee_edge.module [hunk 8]
```diff
@@ -1353,3 +1369,29 @@ function _apigee_edge_set_sync_in_progress(?bool $in_progress = NULL) {
...
+  if (isset($user)) {
+    $args['@user'] = Markup::create($user->getDisplayName());
+    $title = t('@developer_app of @user', $args);
+  }
+  else {
+    $title = t('My @developer_app', $args);
+  }
+
+  // Modules and themes can alter the title.
+  \Drupal::moduleHandler()->alter('my_developer_apps_title', $title, $user);
```
Comment (mxr576) [inline apigee_edge.module]: So as said in our call... `apigee_edge_my_developer_apps_title` aka. `MODULE_NAME_hook`.

**Review by mxr576 — CHANGES_REQUESTED** [review]

File: tests/modules/apigee_edge_test/apigee_edge_test.module [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * @file
+ * Helper module for the apigee_edge tests.
+ */
+
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_my_developer_apps_title_alter().
```
Comment (mxr576) [inline tests/modules/apigee_edge_test/apigee_edge_test.module]: You missed this one :P

**Review by mxr576 — CHANGES_REQUESTED** [review]

**Review by mxr576 — APPROVED** [review]

**Comment by peterserfozo** [comment]
Test: https://travis-ci.org/peterserfozo/apigee-edge-drupal/builds/411232200

**Pull request merged**